File: 1-Basics/1.2-Basic_Maths/solution_in_python/4-Palindrome_Number.py
def check_palindrome(n):
    if(n<0):
        return False
    
    num = 0
    rem = n

    while(rem != 0):
        last = rem%10
        num = (num*10) + last
        rem = rem//10

    return (num==n)


n = int(input("Enter Digit N : "))

print(check_palindrome(n))

File: 1-Basics/1.2-Basic_Maths/solution_in_cpp/4-Palindrome_Number.cpp
#include<bits/stdc++.h>
using namespace std;

bool Palindrome(int n){
    if(n<0) return false;

    int num = 0;
    int rem = n;
    while(rem != 0){
        int last = rem %10;
        num = (num*10) + last;
        rem = rem /10;
    }

    return (num==n);
}

int main(){
    cout<<"Enter Number N : ";
    int n;
    cin>>n;

    cout<<(Palindrome(n) ? "true" : "false");

    return 0;
}